src/common/feedGenerator.ts

```diff
@@ -77,10 +79,26 @@ export const feedToFilters = async (
             'adv.id = fas."advancedSettingsId" AND fas."feedId" = :feedId',
             { feedId },
           )
-          .where('COALESCE(fas.enabled, adv.defaultEnabledState) = false')
-          .getQuery();
+          .where('COALESCE(fas.enabled, adv.defaultEnabledState) = false');
+
+        const experimentSettings = features?.advanced_settings_default_values;
+        if (!experimentSettings?.enabled) {
+          return `s.advancedSettings && array(${subQuery.getQuery()})`;
+        }
```

Comment:
I suggest changing this control flow. Instead of returning earlier as you did, we can update the subquery only if the experiment settings are enabled and then we will have a single return statement.

Reply:
That's a good way to do it. I agree it will make the result much more predictable.